cisagov/ScubaGear
# Fix MS.TEAMS.1.5 Policy of ScubaGear

#  #

## 🗣 Description ##
Currently ScubaGear (in Emerald) shows incorrect results for the MS.TEAMS.1.5v1 - with Criticality and Details columns of report showing undeclared values. The objective of this policy is to check that dial-in users are not configured to be auto admitted into meetings and it is applicable to all policies. The 'who can bypass the lobby' setting tested through another teams policy (1.3) auto selects the true/false value for the dial-in users. But there are some configuration paths that can leave this setting (dial-in users to bypass lobby) to be in true mode. This policy enforces and checks this setting to be false irrespective of previous setting and for all meeting policies. So the implementation is: for all meeting policies, check that 'dial-in users can bypass the lobby' is set to false and count any policies that are non-compliant. 


## 💭 Motivation and context ##
closes #452 

## 🧪 Testing ##
Tested the feature branch with G5 and E5 tenants with following test cases - all passed:

In Teams -> Meeting Policies -> Meeting Join and Lobby, do the following:
1. Set "Who can bypass the lobby" to one of: People in my org, People in my org and guests, People in my org, trusted org, and guests.
2. Set "People dialingin can bypass the lobby" to Off

Above 3 tests should pass the policy 1.5. For negative tests,  set "People dialingin can bypass the lobby" to On and make sure that policy 1.5 fails with a detail that "either everyone or dial-in users are allowed to bypass the lobby". Test this for a custom policy as this policy is applicable to all policies. 

## ✅ Pre-approval checklist ##

      in code comments.
      to reflect the changes in this PR.

## ✅ Pre-merge checklist ##


## ✅ Post-merge checklist ##
